core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -2521,33 +2521,19 @@ class DAGSchedulerSuite extends SparkFunSuite with LocalSparkContext with TimeLi
       (Success, makeMapStatus("hostD", 2))))
     assert(mapOutputTracker.findMissingPartitions(shuffleId2) === Some(Seq.empty))
 
+    // Simulate the scenario of executor lost
+    runEvent(ExecutorLost("exec-hostC", ExecutorKilled))
```

Comment:
The behavior difference between 2.3 and 2.4 is related to #21758, which move the output status clean work forward: https://github.com/apache/spark/pull/21758/files#diff-6a9ff7fb74fd490a50462d45db2d5e11L1390.
So I fix the behavior by simulating the executor lost because here we want a scenario of missing some partitions while rerunning the shuffle map stage.